test/assets.js

```diff
@@ -0,0 +1,523 @@
+
+    it('should populate the assets for the utxo provided', (done) => {
+      controller._populateUtxosAssets(utxosToPopulate)
+        .then(() => {
+          should(utxosToPopulate[0]).have.property('assets');
+          should(utxosToPopulate[1]).have.property('assets');
+          should(utxosToPopulate[2]).have.property('assets');
+          done();
+        })
+        .catch(done);
+    });
+  });
+
+  describe('_loadBestUtxos', () => {
+    const node = {
+      services: {
+        bitcoind: {
+          height: 534203,
+        },
+      },
+    };
+    const handle = {
+      utxosIterator: 0,
+      valueNeeded: 25000,
+      includeAssets: false,
+      choosenUtxos: [],
+    };
+
+    describe('when the first one is enough', () => {
+      const handleToTest = Object.assign({}, handle);
+      const controller = new AssetController(node);
+      handleToTest.utxos = [ { value: 30000 }, { value: 15000 } ];
+      it('should get the higher value utxo only', (done) => {
+        controller._loadBestUtxos(handleToTest)
+          .then((neededUtxos) => {
+            should(neededUtxos.length).be.exactly(1);
+            done();
+          })
+          .catch(done);
+      });
+    });
+
+    describe('when two are needed', () => {
+      const handleToTest = Object.assign({}, handle);
+      handleToTest.valueNeeded = 40000;
+      handleToTest.utxos = [ { value: 30000 }, { value: 15000 } ];
+      const controller = new AssetController(node);
+      it('should get the first two utxo with higher value', (done) => {
+        controller._loadBestUtxos(handleToTest)
+          .then((neededUtxos) => {
+            should(neededUtxos.length).be.exactly(2);
+            done();
+          })
+          .catch(done);
+      });
+    });
+
+    describe('when more are needed', () => {
+      const handleToTest = Object.assign({}, handle);
+      handleToTest.valueNeeded = 60000;
+      handleToTest.utxos = [ { value: 30000 }, { value: 20000 }, { value: 15000 } ];
+      const controller = new AssetController(node);
+      it('should get the first three utxo with higher value', (done) => {
+        controller._loadBestUtxos(handleToTest)
+          .then((neededUtxos) => {
+            should(neededUtxos.length).be.exactly(3);
+            done();
+          })
+          .catch(done);
+      });
+    });
+
+    describe('when there are not utxos available', () => {
+      const handleToTest = Object.assign({}, handle);
+      handleToTest.utxos = [];
+      const controller = new AssetController(node);
+      it('should throw insufficient funds error', (done) => {
+        controller._loadBestUtxos(handleToTest)
+          .then(() => {
+            done('should throw not enough funds');
+          })
+          .catch((err) => {
+            should(err.message).be.exactly('Not enough funds to make the transaction');
+            done();
+          });
+      });
+    });
+  });
+
+  describe('_loadMinimumUtxosCycle', () => {
```

Comment:
It makes use of the actual implementations _loadBestUtxos, buildFunc, and _evaluateInputsValue, testing more than one implementation, it would be better to use a mock

Reply:
Add tests to validate the purpose of the method that is to perform cycles